src/routers/__tests__/SwitchRouter-test.js

```diff
@@ -4,184 +4,149 @@ import React from 'react';
-    const router = getExampleRouter();
-    const state = router.getStateForAction({ type: NavigationActions.INIT });
-    const state2 = router.getStateForAction(
-      { type: NavigationActions.NAVIGATE, routeName: 'A2' },
-      state
-    );
-    expect(state2.routes[0].index).toEqual(1);
-    expect(state2.routes[0].routes.length).toEqual(2);
+    const { navigateTo, getState } = getRouterTestHelper(getExampleRouter());
 
-    const state3 = router.getStateForAction(
-      { type: NavigationActions.NAVIGATE, routeName: 'B' },
-      state2
-    );
+    navigateTo('A2');
+    expect(getState().routes[0].index).toEqual(1);
+    expect(getState().routes[0].routes.length).toEqual(2);
 
-    expect(state3.routes[0].index).toEqual(0);
-    expect(state3.routes[0].routes.length).toEqual(1);
+    navigateTo('B');
+    expect(getState().routes[0].index).toEqual(0);
+    expect(getState().routes[0].routes.length).toEqual(1);
   });
 
   test('does not reset the route on unfocus if resetOnBlur is false', () => {
-    const router = getExampleRouter({ resetOnBlur: false });
-    const state = router.getStateForAction({ type: NavigationActions.INIT });
-    const state2 = router.getStateForAction(
-      { type: NavigationActions.NAVIGATE, routeName: 'A2' },
-      state
+    const { navigateTo, getState } = getRouterTestHelper(
+      getExampleRouter({ resetOnBlur: false })
     );
-    expect(state2.routes[0].index).toEqual(1);
-    expect(state2.routes[0].routes.length).toEqual(2);
 
-    const state3 = router.getStateForAction(
-      { type: NavigationActions.NAVIGATE, routeName: 'B' },
-      state2
-    );
+    navigateTo('A2');
+    expect(getState().routes[0].index).toEqual(1);
+    expect(getState().routes[0].routes.length).toEqual(2);
 
-    expect(state3.routes[0].index).toEqual(1);
-    expect(state3.routes[0].routes.length).toEqual(2);
+    navigateTo('B');
+    expect(getState().routes[0].index).toEqual(1);
+    expect(getState().routes[0].routes.length).toEqual(2);
   });
 
   test('ignores back by default', () => {
-    const router = getExampleRouter();
-    const state = router.getStateForAction({ type: NavigationActions.INIT });
-    const state2 = router.getStateForAction(
-      { type: NavigationActions.NAVIGATE, routeName: 'B' },
-      state
+    const { navigateTo, back, getState } = getRouterTestHelper(
+      getExampleRouter()
     );
-    expect(state2.index).toEqual(1);
 
-    const state3 = router.getStateForAction(
-      { type: NavigationActions.BACK },
-      state2
-    );
+    navigateTo('B');
+    expect(getState().index).toEqual(1);
 
-    expect(state3.index).toEqual(1);
+    back();
+    expect(getState().index).toEqual(1);
   });
 
   test('handles initialRoute backBehavior', () => {
-    const router = getExampleRouter({ backBehavior: 'initialRoute' });
+    const { navigateTo, back, getState } = getRouterTestHelper(
+      getExampleRouter({ backBehavior: 'initialRoute', initialRouteName: 'B' })
```

Comment:
added initialRouteName param in router here, because the test does not ensure this param is actually wired to anything and could only fallback to first route instead of using the param